### PR TITLE
Record time taken to complete review

### LINKED ISSUE #16: Record time to complete a review
Currently the form records the timestamp at the end of the submission [here](https://github.com/lacabra/validate-digitalpublicgoods/blob/e9b093748842c84b17cbdbafc7ddd2301da36820/components/openpr.js#L193).

In addition to the current timestamping, we would like to record the time difference between the moment that the user starts reviewing the first question to the time they submit, (recorded as a `Time_end - Time_start` in milliseconds) and added as an additional field to the submission. We will later use this additional data point to filter out those submissions where the user quickly browsed the entirety of the form arbitrarily clicking `yes` or `no` without thoughtfully analyzing each question.

### PR DESCRIPTION
**Pull request by rachita19082**
This pull request resolves issue #16

### CODE REVIEW

**Comment by rachita19082** [comment]
@lacabra I have made some changes to the code which I think will resolve the issue. However, I was unable to figure out a way to test the code. Please let me know if there is any way for me to test the code and if any changes are required. While committing, I also got the error _"husky > pre-commit hook failed (add --no-verify to bypass)"_, can you let me know what could be the reason behind this?

**Comment by lacabra** [comment]
@rachita19082: I have expanded the README documentation in 48ba60a, refer to the [Development Environment](https://github.com/lacabra/validate-digitalpublicgoods#-development-environment) section.

If you follow the updated instructions, your local environment will open PRs upon submit of the form to your clone of the publicgoods-candidates repos (instead of the UNICEF one), and that is the way how you can test whether your changes work as intended. Kindly link a successful PR to your clone to show that the functionality works as intended.

**Comment by lacabra** [comment]
@rachita19082: the precommit husky check validates the linting of your files. You should see the actual error by running:
```bash
npm run lint
```
which you can fix manually, or automatically by running:
```
npm run lint:fix
```

You can typically find this information in `package.json` ;)

**Comment by rachita19082** [comment]
@lacabra I have fixed the linting issue and corrected a few mistakes in my code. However, I am facing problems with the OAuth app. What needs to be added to the Homepage url and authorization callback URL? Currently, I have added http://localhost:3000 as the Homepage url and http://localhost:3000/callback as the callback URL but it doesn't seem to work.

**Comment by rachita19082** [comment]
@lacabra I have tested the functionality of the code in this [PR](https://github.com/rachita19082/publicgoods-candidates-test/pull/1/commits/951c3e0c322b5efda09f4a20facd3540e3790299). The time is shown in seconds. Please let me know if any other changes are needed.

File: pages/projects.js
Comment (lacabra) [inline pages/projects.js]: why do you use `start_time` here? for consistency, I'd rather use `startTime={startTime}` which should be perfectly fine. Curious to hear your rationale for switching between one and the other?

File: pages/review.js
Comment (lacabra) [inline pages/review.js]: If you change the one above, then again here you would use `startTime={props.startTime}`. Given that all the codebase uses camelCase, I see no reason to break the consistency here :)

**Review by lacabra — CHANGES_REQUESTED** [review]

**Comment by lacabra** [comment]
@rachita19082, the linter was added to the CI recently, and it now fails. Kindly take a look, and make sure that the `CI/build` passes before this pull request can be merged. Thank you

File: pages/review.js
Comment (rachita19082) [reply pages/review.js]: @lacabra I try to avoid using the same name for the parameter and the state because I tend to get confused about what is being referenced and sometimes it has led to errors as well. But I agree that here it is better to keep the names consistent. I have made the required changes in commit a657fd5.

**Comment by rachita19082** [comment]
> @rachita19082, the linter was added to the CI recently, and it now fails. Kindly take a look, and make sure that the `CI/build` passes before this pull request can be merged. Thank you

I have fixed the linting issues but the build requires approval. Please approve and I will make the necessary changes if the build fails again.

**Review by lacabra — APPROVED** [review]
Thanks @rachita19082, everything looks good now! 👍

**Pull request merged**
